Makefile

```diff
@@ -575,10 +575,17 @@ package:  ## Build rpm packages
 	rm -rf build/* *.src.rpm ~/rpmbuild/RPMS/*
 	./contrib/build_rpm.sh
 
+build/podman.repo: package
+
 # Remember that rpms install exec to /usr/bin/podman while a `make install`
 # installs them to /usr/local/bin/podman which is likely before. Always use
 # a full path to test installed podman or you risk to call another executable.
-package-install: package  ## Install rpm packages
-	sudo ${PKG_MANAGER} -y install ${HOME}/rpmbuild/RPMS/*/*.rpm
+package-install: package build/podman.repo  ## Install rpm packages
+	#sudo ${PKG_MANAGER} -y install ${HOME}/rpmbuild/RPMS/*/*.rpm
+	sudo cp -f build/podman.repo /etc/yum.repos.d/podman.repo
+	sudo ${PKG_MANAGER} -y --enablerepo podman install podman podman-remote
```

Comment:
Using sudo in a makefile leaves me slightly uncomfortable as it assumes a lot.  For example, if this target is running inside a container as root, it will fail if sudo isn't installed (perhaps for security reasons, perhaps just to keep the image small).  Would it make sense to leave out the `sudo` and just rely on the user running `sudo make package-install` when required?

Reply:
Yes I agree, I don't think sudo should be in the Makefile.

Reply:
I will address all comments receives so far, but before implementing these changes I need help addresing the issue i mentioned at https://github.com/containers/libpod/pull/4815#issuecomment-572085796 -- so far in 2/4 platforms yum/dnf is unable to find the rpms from the builder repo.

Reply:
@rhatdan Calling make with sudo would break a huge number of things, think about temp files created as root. 

While is technically easy to detect root and fail if you did not call it as root, it would create a never ending list of problems as all make targets run as root will produce artifacts with wrong user. I really do not with to create any file owned by root in a non-root user home directory.

The jobs we run on rdo now are expecting to find a repository on `project_root/build/builset` and I would prefer not to have to alter them to look in different location.

Look inside the makefile and you will see that `package-install` target depends-on `package` for good reasons. This means that if user calls it with `sudo`, both targets will run as root. build_rpm.sh is also calling copr makefile which produces the source rpms innside build/ folder, so it would introduce root owned files to current project_root folder.

I am ok to remove sudo from makefile if we find a way to avoid messing build when someone runs `sudo make <random-target>` by mistake. I personally find more annoying the messed file ownership that the feature of not calling sudo exactly where is needed.

Maybe we should add a protection for each target and prevent it from running other than how is inteded (like some kind of ensure-root and ensure-non-root checks)?